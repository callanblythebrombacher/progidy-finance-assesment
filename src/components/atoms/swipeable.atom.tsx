import React from 'react';
import {Swipeable} from 'react-native-gesture-handler';
import {SwipeableComponentProps} from '../../interfaces/atom.interfaces.ts'; // Remove '.ts' extension

export const SwipeableComponent = React.forwardRef<
  any,
  SwipeableComponentProps
>(({children, renderRightActions, friction, rightThreshold}, ref) => {
  return (
    <Swipeable
      ref={ref}
      renderRightActions={renderRightActions}
      friction={friction ?? 2}
      rightThreshold={rightThreshold ?? 40}>
      {children}
    </Swipeable>
  );
});
