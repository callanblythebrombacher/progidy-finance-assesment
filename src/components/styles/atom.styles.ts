import {StyleSheet} from 'react-native';

export const autoCompleteStyles = StyleSheet.create({
  autoCompleteListItem: {
    backgroundColor: 'white',
    borderBottomColor: '#e7e7e7',
    borderBottomWidth: 1,
    height: 60,
  },
  autocompleteListContainer: {
    position: 'absolute',
    width: '100%',
    top: 65, // Adjust as needed based on your layout
    backgroundColor: '#fff',
    borderRadius: 5,
    elevation: 5, // Android shadow
    shadowColor: '#000', // iOS shadow
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },
});

export const tableStyles = StyleSheet.create({
  tableHeader: {backgroundColor: '#dddddd', width: '100%'},
  tableRow: {height: 70, backgroundColor: 'white'},
});

export const pickerStyles = StyleSheet.create({
  container: {
    zIndex: 3,
    marginLeft: 10,
    marginRight: 10,
    marginBottom: 2,
  },
});
