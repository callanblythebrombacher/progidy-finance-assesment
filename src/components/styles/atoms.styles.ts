import {StyleSheet} from 'react-native';

export const autoCompleteStyles = StyleSheet.create({
  item: {
    backgroundColor: 'white',
    borderBottomColor: '#e7e7e7',
    borderBottomWidth: 1,
    height: 60,
  },
  container: {
    position: 'absolute',
    top: 61, // Adjust as needed based on your layout
    backgroundColor: '#fff',
    borderRadius: 5,
    elevation: 5, // Android shadow
    shadowColor: '#000', // iOS shadow
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.8,
    shadowRadius: 2,
    left: -10,
  },
});

export const tableStyles = StyleSheet.create({
  tableHeader: {backgroundColor: '#dddddd', width: '100%'},
  tableRow: {height: 70, backgroundColor: 'white'},
});

export const pickerStyles = StyleSheet.create({
  container: {
    zIndex: 3,
    marginLeft: 10,
    width: 140,
  },
});
