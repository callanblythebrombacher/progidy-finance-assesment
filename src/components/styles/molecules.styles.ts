import {StyleSheet} from 'react-native';
export const tableStyles = StyleSheet.create({
  table: {
    position: 'relative',
    zIndex: 1,
    top: 10,
  },
  scrollView: {
    backgroundColor: 'white',
  },
  deleteButtonView: {
    backgroundColor: 'red',
    width: 60,
    justifyContent: 'center',
    alignItems: 'center',
  },
  pagination: {
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export const searchBarStyles = StyleSheet.create({
  container: {
    backgroundColor: '#ffffff',
    width: '100%',
    zIndex: 4,
  },
  input: {
    height: 50,
    borderWidth: 1,
    borderColor: '#000000',
    padding: 10,
    marginBottom: 20,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 10,
    backgroundColor: 'white',
    borderRadius: 8,
  },
  iconButton: {
    position: 'absolute',
    right: 3,
    top: 9,
  },
});
