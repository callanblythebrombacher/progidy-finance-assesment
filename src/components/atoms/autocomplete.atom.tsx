import React from 'react';
import {FlatList, View} from 'react-native';
import {List} from 'react-native-paper';
import {autoCompleteStyles as styles} from '../styles/atoms.styles';
import {AutoCompleteProps} from '../../interfaces/atom.interfaces.ts';
import useScreenDimensions from '../../hooks/useScreenDimensions';

/**
 * Autocomplete component displays a list of items based on autocompleteData.
 * Users can select an item, triggering onItemPress.
 */
export const Autocomplete: React.FC<AutoCompleteProps> = ({
  autocompleteData,
  onItemPress,
}) => {
  const {height, width} = useScreenDimensions(); // Hook to get screen dimensions

  /**
   * Renders each item in the FlatList.
   * @param item - Item object containing listTitle and onPress function.
   */
  const renderItem = ({
    item,
  }: {
    item: {listTitle: string; onPress: () => void};
  }) => (
    <List.Item
      style={styles.item}
      title={item.listTitle}
      onPress={() => {
        onItemPress(item.listTitle); // Trigger onItemPress with selected item's title
        item.onPress(); // Execute item-specific onPress function
      }}
    />
  );

  return (
    <View
      style={[
        styles.container,
        {
          // Adjust container height based on whether autocompleteData is empty or not
          height:
            autocompleteData && autocompleteData.length > 0
              ? height - (height > width ? 100 : 80)
              : 0,
          width: width, // Set container width to screen width
        },
      ]}>
      {/* FlatList to render autocompleteData */}
      <FlatList
        data={autocompleteData}
        renderItem={renderItem} // Render each item using renderItem function
        keyExtractor={(item, index) => index.toString()} // Extract keys for each item
      />
    </View>
  );
};
