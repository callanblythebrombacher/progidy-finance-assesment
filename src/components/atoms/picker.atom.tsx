import React, {useState} from 'react';
import {PickerProps} from '../../interfaces/atom.interfaces.ts';
import DropDownPicker from 'react-native-dropdown-picker';
import {View} from 'react-native';
import {pickerStyles as styles} from '../styles/atoms.styles.ts';

export const PickerComponent: React.FC<PickerProps> = ({
  items,
  setValue,
  value,
}) => {
  const [open, setOpen] = useState(false);
  const [selectItems, setSelectItems] = useState(items);

  return (
    <View style={styles.container}>
      <DropDownPicker
        open={open}
        value={value}
        items={selectItems}
        setOpen={setOpen}
        setValue={setValue}
        setItems={setSelectItems}
      />
    </View>
  );
};
