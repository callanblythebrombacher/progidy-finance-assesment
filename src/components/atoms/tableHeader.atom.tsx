import React from 'react';
import {DataTable} from 'react-native-paper';
import {tableStyles as styles} from '../styles/atoms.styles.ts';
import {TableHeaderProps} from '../../interfaces/atom.interfaces.ts';
export const TableHeader: React.FC<TableHeaderProps> = ({headerData}) => {
  return (
    <DataTable.Header style={styles.tableHeader}>
      {headerData.map(({item, isNumeric}, index) => (
        <DataTable.Title key={index} numeric={isNumeric ? true : undefined}>
          {item}
        </DataTable.Title>
      ))}
    </DataTable.Header>
  );
};
