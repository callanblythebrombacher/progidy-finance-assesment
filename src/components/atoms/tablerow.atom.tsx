import React from 'react';
import {DataTable} from 'react-native-paper';
import {tableStyles as styles} from '../styles/atom.styles.ts';
import {TableRowProps} from '../interfaces/atom.interfaces.ts';

export const TableRow: React.FC<TableRowProps> = ({cellData}) => {
  return (
    <DataTable.Row style={styles.tableRow}>
      {cellData.map(({item, isNumeric}, index) => (
        <DataTable.Cell key={index} numeric={isNumeric ? true : undefined}>
          {item}
        </DataTable.Cell>
      ))}
    </DataTable.Row>
  );
};
