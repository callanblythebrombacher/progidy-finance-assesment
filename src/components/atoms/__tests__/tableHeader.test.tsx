import React from 'react';
import {render} from '@testing-library/react-native';
import {TableHeader} from '../tableHeader.atom';
import {tableStyles as styles} from '../../styles/atoms.styles';

import {describe, it, expect} from '@jest/globals';

describe('TableHeader Atom', () => {
  const mockHeaderData = [
    {item: 'Name', isNumeric: false},
    {item: 'Age', isNumeric: true},
    {item: 'City', isNumeric: false},
  ];

  it('renders correctly', () => {
    const {toJSON} = render(<TableHeader headerData={mockHeaderData} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders DataTable.Header component', () => {
    const {getByTestId} = render(<TableHeader headerData={mockHeaderData} />);
    const dataTableHeader = getByTestId('data-table-header');
    expect(dataTableHeader).toBeTruthy();
    expect(dataTableHeader.props.style).toEqual(styles.tableHeader);
  });

  it('renders DataTable.Title components with correct props', () => {
    const {getByText} = render(<TableHeader headerData={mockHeaderData} />);

    mockHeaderData.forEach(header => {
      const dataTableTitle = getByText(header.item);
      expect(dataTableTitle).toBeTruthy();
      expect(dataTableTitle.props.numeric).toBe(header.isNumeric);
    });
  });
});
