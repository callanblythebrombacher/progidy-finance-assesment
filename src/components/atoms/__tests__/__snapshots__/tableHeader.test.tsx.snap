// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader Atom renders correctly 1`] = `
<View
  style={
    [
      {
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingHorizontal": 16,
      },
      {
        "borderBottomColor": "rgba(231, 224, 236, 1)",
      },
      {
        "backgroundColor": "#dddddd",
        "height": 50,
        "width": "100%",
      },
    ]
  }
  testID="data-table-header"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 12,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          [
            {
              "alignItems": "center",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 24,
            },
            {
              "maxHeight": 24,
            },
            {},
            {
              "color": "rgba(28, 27, 31, 0.6)",
            },
            undefined,
          ],
        ]
      }
    >
      Name
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 12,
        },
        {
          "justifyContent": "flex-end",
        },
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          [
            {
              "alignItems": "center",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 24,
            },
            {
              "maxHeight": 24,
            },
            {},
            {
              "color": "rgba(28, 27, 31, 0.6)",
            },
            undefined,
          ],
        ]
      }
    >
      Age
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignContent": "center",
          "flex": 1,
          "flexDirection": "row",
          "paddingVertical": 12,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          [
            {
              "alignItems": "center",
              "fontSize": 12,
              "fontWeight": "500",
              "lineHeight": 24,
            },
            {
              "maxHeight": 24,
            },
            {},
            {
              "color": "rgba(28, 27, 31, 0.6)",
            },
            undefined,
          ],
        ]
      }
    >
      City
    </Text>
  </View>
</View>
`;
